shift-react-app/src/steps.ts

```diff
@@ -0,0 +1,80 @@
+import {
+  readFileSync,
+  writeFileSync,
+} from 'fs';
+import { join } from 'path';
+import { spawn } from 'child_process';
+import { EOL } from 'os';
+
+export function setupProxy(): string {
+  const pjsonBuffer = readFileSync('./package.json');
+  let pjson;
+  try {
+    pjson = JSON.parse(pjsonBuffer.toString());
+  } catch (e) {
+    throw new Error('Cannot parse package.json');
+  }
+  const { scripts: { eject } = {} as Record<string, string> } = pjson;
+  if (eject !== 'react-scripts eject') {
+    throw new Error('Can not run outside of a create-react-app or an ejected project');
+  }
+  const proxyFile = join('src', 'setupProxy.js');
+  const proxyCode = `/*
+ * This file was auto-generated by ShiftJS.
+ *
+ * It is used to mock the calls to the backend when developing locally.
+ * Please keep it with your source control tool.
+ * Please do not remove or change it manually.
+ */
+const { setupProxy } = require('@binaris/shift-local-proxy');
+module.exports = setupProxy(__dirname);
+`;
+  writeFileSync(proxyFile, proxyCode, { flag: 'wx' });
+  return `Created ${proxyFile}, please commit this file`;
+}
+
+export function installPackages(): Promise<string> {
+  return new Promise((resolve, reject) => {
+    const dependencies = [
+      '@binaris/shift-local-proxy',
+      '@binaris/shift-babel-macro',
+      '@binaris/shift-fetch-runtime',
+      '@binaris/shift-db',
+    ];
+    const args = [
+      'install',
+      '--save',
+      '--loglevel',
+      'error',
+    ].concat(dependencies);
+    const child = spawn('npm', args, { stdio: 'inherit' });
+    child.on('close', (code) => {
+      if (code !== 0) {
+        reject(new Error(`Could not install ${dependencies.join(' ')}`));
+        return;
+      }
+      resolve(`Packages installed
+Modified package.json, please commit this file`);
+    });
+  });
+}
+
+export function ignoreShift(): string {
+  const gitIgnoreFile = '.gitignore';
+  const ignorePattern = '.shift*';
+  const ignoreText = '# The following line was added by ShiftJS' + EOL + ignorePattern;
+  let initialContent;
+  try {
+    initialContent = readFileSync(gitIgnoreFile, { encoding: 'utf8' });
+  } catch (e) {
+    return `Did not update ${gitIgnoreFile}`;
+  }
+  const lines: string[] = initialContent.split(EOL);
+  const found = lines.find((line) => line && line.replace(/ +$/, '') === ignorePattern);
```

Comment:
Apparently you could run: `git check-ignore 'shift*'`

Reply:
An installed and and available git for the script to run is not assumed and shouldn't be.

Reply:
I wouldn't expect to find a `.gitignore` on a machine that doesn't have git installed but OK.